### PR TITLE
Homework 7 JDI

### PR DESCRIPTION
**Pull request by parfenovaaa**

### CODE REVIEW

File: src/main/java/com/epam/tc/hw7/site/SiteJdi.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import com.epam.tc.hw7.site.section.LoginForm;
+import org.openqa.selenium.WebElement;
+
+public class SiteJdi {
+
+    public static HomePage homePage;
+    public static LoginForm loginForm;
+    public static MetalsColoursPage metalsColoursPage;
+
+    @Css(".profile-photo [ui=label]") public static UIElement userName;
```
Comment (cold-flame) [inline src/main/java/com/epam/tc/hw7/site/SiteJdi.java:17]: In JdiSite class need to declare variable WebPage class

File: src/test/java/com/epam/tc/hw7/BaseTest.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        }
+    }
+
+    @Step("Assert data.")
+
+    public void assertData(String actual, String expected) {
+        Assert.assertEquals(actual, expected);
+    }
+
+    public MetalsColoursData  readJsonToMetalsColoursData(int j) throws IOException, ParseException {
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/BaseTest.java:80]: Why you put this in there ?

File: src/test/java/com/epam/tc/hw7/BaseTest.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            } else {
+                if (oddRadioButtons.selected(String.valueOf(element))) {
+                    continue;
+                } else {
+                    oddRadioButtons.select(String.valueOf(element));
+                }
+            }
+        }
+
+        String[] elements = testData.elements;
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/BaseTest.java:133]: JDI can fill all fields

**Review by cold-flame — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/BaseTest.java (hunk 2 above)
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw7/BaseTest.java]: Moved to class ReadTestDataFromJson

File: src/main/java/com/epam/tc/hw7/site/SiteJdi.java (hunk 1 above)
Comment (parfenovaaa) [reply src/main/java/com/epam/tc/hw7/site/SiteJdi.java:21]: ok

File: src/test/java/com/epam/tc/hw7/BaseTest.java (hunk 3 above)
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw7/BaseTest.java:88]: ok

File: src/test/java/com/epam/tc/hw7/BaseTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        loginForm.loginAs(new LoginUserData());
+        homePage.checkOpened();
+    }
+
+    @Test
+    public void metalsColoursPageTest() {
+        shouldBeLoggedIn();
+        topMenu.select(MetalsColors);
+        metalsColoursPage.shouldBeOpened();
+        ReadTestDataFromJson dataFromJson = new ReadTestDataFromJson();
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/BaseTest.java:59]: You need to use @Test(dataProvider = "name") here

File: src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import java.io.FileReader;
+import java.io.IOException;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+public class ReadTestDataFromJson {
+
+    public MetalsColoursData readJsonToMetalsColoursData(int j) {
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java:12]: name of 'j' is unclear purpose

File: src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+        String fileName = "src/test/resources/JDI_ex8_metalsColorsDataSet.json";
+        JSONObject jsonObject = null;
+        try {
+            jsonObject = (JSONObject) new JSONParser().parse(new FileReader(fileName));
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java:17]: It's bad idea to use that. You call this method in cycle above. Need to change to single open file

**Review by cold-flame — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java (hunk 5 above)
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java]: ok, i understand

File: src/test/java/com/epam/tc/hw7/BaseTest.java (hunk 4 above)
Comment (parfenovaaa) [reply src/test/java/com/epam/tc/hw7/BaseTest.java]: changed ReadTestDataFromJson to data provider class

File: src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+public class ReadTestDataFromJson {
+
+    @DataProvider(name = "readJsonToMetalsColoursData")
+    public static Object[][] readJsonToMetalsColoursData() throws IOException, ParseException {
+
+        String fileName = "src/test/resources/JDI_ex8_metalsColorsDataSet.json";
+        JSONObject jsonObject = (JSONObject) new JSONParser().parse(new FileReader(fileName));
+
+        Object[][] returnValue = new Object[jsonObject.size()][6];
```
Comment (cold-flame) [inline src/test/java/com/epam/tc/hw7/ReadTestDataFromJson.java:19]: You mustn't use '6' in this case. Try to parameterize this or develop new structure

**Review by cold-flame — CHANGES_REQUESTED** [review]

**Review by cold-flame — APPROVED** [review]

**Pull request merged**